### ISSUE TITLE
New Rule: `no-string-module-specifier`

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#16332]
### Rule details

Restricts module specifiers written as string literals

### Related ECMAScript feature

[Arbitrary module namespace identifier names](https://github.com/tc39/ecma262/pull/2154)

### What type of rule is this?

Enforces a formatting/stylistic preference

### Example code

```js
// invalid
import { "a" as foo } from "module";
export { "a" as foo } from "module";
export { foo as "a" } from "module";
export { "a" as "a" } from "module";

// valid
import { " 👶 " as foo } from "module";
export { " 👶 " as foo } from "module";
export { foo as " 👶 " } from "module";
export { " 👶 " as " 👶 " } from "module";
```

### Why should this rule be in the core instead of a plugin?

ES2022 allows arbitrary strings to be used for import specifier and export specifier. (ref: https://github.com/eslint/eslint/issues/15465)

With this feature, module specifiers that are valid as ordinary JavaScript identifiers can also be written as string literals.

```js
import { "a" as foo } from "mod";
```

```js
import { a as foo } from "mod";
```

These two import statements mean exactly the same thing.　Such a code would confuse code readers.

I think we need rules that can make consistent this.

### Participation

- [X] I am willing to submit a pull request to implement this rule.

### Additional comments

Need a better name for the rule?

**Comment by nzakas** [comment]
Thanks for the suggestion. Even though this is new syntax, we aren't adding new rules that just disallow some syntax. If this is important to you, you can still create a custom rule to enforce this stylistic preference. You can also use `no-restricted-syntax` to just disallow what you'd like.
